src/Plugins/RestServer/Helpers/ScriptHelper.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ScriptHelper.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Linq;
+
+namespace Neo.Plugins.RestServer.Helpers
+{
+    internal static class ScriptHelper
+    {
+        public static bool InvokeMethod(ProtocolSettings protocolSettings, DataCache snapshot, UInt160 scriptHash, string method, out StackItem[] results, params object[] args)
+        {
+            using var scriptBuilder = new ScriptBuilder();
+            scriptBuilder.EmitDynamicCall(scriptHash, method, CallFlags.ReadOnly, args);
+            byte[] script = scriptBuilder.ToArray();
+            using var engine = ApplicationEngine.Run(script, snapshot, settings: protocolSettings, gas: RestServerSettings.Current.MaxGasInvoke);
+            results = engine.State == VMState.FAULT ? [] : [.. engine.ResultStack];
+            return engine.State == VMState.HALT;
+        }
+
+        public static ApplicationEngine InvokeMethod(ProtocolSettings protocolSettings, DataCache snapshot, UInt160 scriptHash, string method, ContractParameter[] args, Signer[]? signers, out byte[] script)
+        {
+            using var scriptBuilder = new ScriptBuilder();
+            scriptBuilder.EmitDynamicCall(scriptHash, method, CallFlags.ReadOnly, args);
```

Comment:
You set CallFlags always ReadOnly, this doesn't make sense.

Reply:
OK ill change that. This was written way back when i didnt know much.